frontend/src/components/profile/ProfileContent.js

```diff
@@ -0,0 +1,240 @@
+    let data
+    try {
+      data = serialize(formData, user)
+    } catch(err) {
+      setState({ message: err.message, success: false })
+      return
+    }
+    setState({ message: undefined, success: undefined })
+    updateSelf(data)
+      .then(() => {
+        setFormErrors({})
+        setState({ message: 'Profile updated', success: true })
+      })
+      .catch(err => {
+        console.log(err)
+        setFormErrors(err.data || {})
+      })
+  }
+
+  /*
+   * Render
+   */
+
+  const props = name =>
+    !formErrors[name] ? { name } : {
+      name,
+      hasFeedback: true,
+      validateStatus: 'error',
+      help: formErrors[name],
+    }
+
+  const errors = []
+    .concat(formErrors?.non_field_errors)
+    .concat(formErrors?.detail)
+    .concat(error?.message)
+    .concat(state.success === false ? state.message : [])
+    .filter(Boolean)
+
+  return (
+    <>
+      <AppPageHeader
+        title='Profile'
+        onBack={() => history.goBack()}
+      />
+      <PageContent>
+        <div>
+          <Title level={2}><UserOutlined /> {user?.username}</Title>
+          <Form
+            key={key}
+            labelCol={{ span: 4 }}
+            wrapperCol={{ span: 12 }}
+            layout="horizontal"
+            autoComplete="off"
+            initialValues={formData}
+            onValuesChange={onValuesChange}
+            onFinish={onSubmit}
+          >
+            {/* Google Chrome is very dumb and tries to autocomplete things
+              even when we explicitely set autocomplete="new-password". Thus
+              we create fake fields, so our real fields stay pristine.
+              The tabIndex & placeholder values are filled to make it keyboard
+              inaccessible. */}
+            <input
+              type="password"
+              autoComplete="password"
+              placeholder="Fake password field"
+              style={hiddenField}
+              tabIndex="-1"
+            />
+
+            <Form.Item label="Email">
+              {user?.email}
+            </Form.Item>
+            <Form.Item
+              label="Password"
+              {...props("password")}
+            >
+              <Input
+                type="password"
+                autoComplete="new-password"
+              />
+            </Form.Item>
+            <Form.Item
+              label="(again)"
+              name="passwordValidation"
+            >
+              <Input
+                type="password"
+                autoComplete="new-password"
+              />
+            </Form.Item>
+            <Form.Item label="First Name" {...props("first_name")} rules={requiredRules}>
+              <Input />
+            </Form.Item>
+            <Form.Item label="Last Name" {...props("last_name")} rules={requiredRules}>
+              <Input />
+            </Form.Item>
+            <Form.Item label="Groups">
+              {groups.map(Options.renderGroup).map(o => <Tag key={o.value}>{o.label}</Tag>)}
```

Comment:
This seem to display all groups regardless of them being associated to the user.

Reply:
Fixed